distributedlog-client/src/main/java/com/twitter/distributedlog/client/serverset/DLZkServerSet.java

```diff
@@ -33,19 +36,43 @@
  */
 public class DLZkServerSet {
 
+    private static final Logger logger = LoggerFactory.getLogger(DLZkServerSet.class);
+
     static final String ZNODE_WRITE_PROXY = ".write_proxy";
 
     private static String getZKServersFromDLUri(URI uri) {
         return uri.getAuthority().replace(";", ",");
     }
 
+    private static Iterable<InetSocketAddress> getZkAddresses(URI uri) {
+        String zkServers = getZKServersFromDLUri(uri);
+        String[] zkServerList = StringUtils.split(zkServers, ',');
+        InetSocketAddress[] zkAddresses = new InetSocketAddress[zkServerList.length];
+        int i = 0;
+        for (String zkServer : zkServerList) {
+            String[] hostAndPort = StringUtils.split(zkServer, ':');
+            int port = 2181;
+            String host = hostAndPort[0];
+            if (hostAndPort.length == 2) {
+                try {
+                    port = Integer.parseInt(hostAndPort[1]);
+                } catch (NumberFormatException nfe) {
+                    logger.warn("Failed to retrieve zookeeper server port from {}. Use default port 2181.",
+                            zkServer, nfe);
+                }
+            }
+            InetSocketAddress address = InetSocketAddress.createUnresolved(host, port);
+            zkAddresses[i] = address;
+            ++i;
```

Comment:
you can get rid of the i index and replace the array with an ArrayList appending as you go. ImmutableList.copyOf can take a Collection